packages/integrations/netlify/README.md

```diff
@@ -72,6 +72,56 @@ export default defineConfig({
 });
 ```
 
+### Run middleware in Edge Functions
+
+When deploying to Netlify Functions, you can choose to use an Edge Function to run your Astro middleware.
+
+To enable this, set the `build.excludeMiddleware` Astro config option to `true`:
+
+```js ins={9}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+import netlify from '@astrojs/netlify/functions';
+
+export default defineConfig({
+  output: 'server',
+  adapter: netlify(),
+  build: {
+    excludeMiddleware: true,
+  },
+});
+```
+
+#### Pass edge context to your site
+
+Netlify Edge Functions provide a [context object](https://docs.netlify.com/edge-functions/api/#netlify-specific-context-object) including metadata about the request, such as a user’s IP, geolocation data, and cookies.
+
+To expose values from this context to your site, create a `netlify-edge-middleware.ts` (or `.js`) file in your project’s [source directory](https://docs.astro.build/en/reference/configuration-reference/#srcdir). This file must export a function that returns the data to add to [Astro’s `locals` object](https://docs.astro.build/en/guides/middleware/#locals), which is available in middleware and Astro routes.
+
+In this example, `visitorCountry` and `hasEdgeMiddleware` would both be added to the Astro’s `locals` object:
+
+```ts
+// src/netlify-edge-middleware.ts
+import type { Context } from 'https://edge.netlify.com';
+
+export default function ({ request, context }: { request: Request; context: Context }) {
```

Comment:
```suggestion
export default function ({ request, context }: { request: Request; context: Context }): object {
```

Reply:
I removed this because I thought `object` was considered bad practice in TypeScript — but now I’m seeing TypeScript ESLint actually relaxed this in v5: https://github.com/typescript-eslint/typescript-eslint/pull/3818

I’m still not sure it adds much though? `object` gives you very weak type checking. `Record<string, unknown>` would be slightly better (prevents you returning a `Map` or a function for example):

```suggestion
export default function ({
  request,
  context,
}: {
  request: Request;
  context: Context;
}): Record<string, unknown> {
```

But not sure it’s worth letting Prettier explode one line into seven 😅

Reply:
I don't have strong opinions. The most important thing is that we should communicate to the user that they can type the retuned type as they want, as long as the returned type is an object :)

Reply:
OK, cool. I think we have it covered then. We say both “a function that returns the data to add to Astro’s `locals` object” above and “must return a JSON-serializable object” below the code sample, plus include a comment inside the code sample about returning serializable data, so I think that should be enough.